src/services/wishlist/addToWishlist.js

```diff
@@ -0,0 +1,24 @@
+import axios from "axios"
+
+//ADD TO WISHLIST
+export const addToWishlist = (product,user,wishlist,setWishlist) => {
+    wishlist.find((prod) => prod.id === product.id)
+      ? alert("Already in wishlist")
+      : 
+      (async () => {
+          try {
+            const { data } = await axios.post(
+              "api/user/wishlist",
+              { product },
+              {
+                headers: {
+                  authorization: user.tokenVal,
+                },
+              }
+            );
+            setWishlist(data.wishlist);
```

Comment:
Since this is a service, if you could make it work without passing setWishlist here it would be better.

Reply:
Thanks for suggestions